BalancedBinaryTree.java

```diff
@@ -0,0 +1,58 @@
+// Time Complexity :O(N) N = no of nodes
+// Space Complexity :O(H) H = height
```

Comment:
Time: O(n^2), The tree is traversed repeatedly to compute height, Please try Bottom-Up Approach as well.